### ISSUE TITLE
[BITV] 9.3.3.2/3.4 - Sharing menu: The "Note to recipient" input field does not have a visual label, it only has a placeholder element. (2)

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/server#37088]
see also https://github.com/nextcloud/server/issues/37091

- [x] https://github.com/nextcloud-libraries/nextcloud-vue/pull/4448
- [x] https://github.com/nextcloud/server/pull/40430

<img width="229" alt="6ae04ea919f771fe600de551bf23aa2a" src="https://user-images.githubusercontent.com/1315170/223563090-3077b30d-d5b6-4c11-b8d4-714cbfc23324.png">

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-9f92015c98-v3-n4

</details>

- [x] 9.3.3.2/10.4 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-9f92015c98-v10-n4

**Comment by jancborchardt** [comment]
@Pytal @JuliaKirschenheuter for the sharing flow redesign we have a pattern of checkboxes with label which open an input or textarea below: https://github.com/nextcloud/server/pull/39472

Question there is: Can we use the same label that is used for the input field also for the input / textarea below? They belong together, and otherwise duplication might be needed. Or which approach would you recommend here?

**Comment by jancborchardt** [comment]
Also cc @Fenn-CS @artonge on the previous comment as they work on that pull request currently.

**Comment by Pytal** [comment]
> @Pytal @JuliaKirschenheuter for the sharing flow redesign we have a pattern of checkboxes with label which open an input or textarea below: #39472 – same pattern as in the above screenshot essentially.
> 
> Question there is: Can we use the same label that is used for the input field also for the input / textarea below? They belong together, and otherwise duplication might be needed. Or which approach would you recommend here?

Good point, since NcCheckboxRadioSwitch contains a label and an input already and a label can only be `for` a single id (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attributes) then I believe we could use `aria-labelledby` here

```
span
  input id="foo"
  label for="foo" id="bar"
span
textarea aria-labelledby="bar"
```

Does that work @michaelnissenbaum ?

**Comment by ShGKme** [comment]
> Good point, since NcCheckboxRadioSwitch contains a label and an input already and a label can only be `for` a single id (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attributes) then I believe we could use `aria-labelledby` here

I checked it with NVDA screen reader, and it works fine.

**Comment by michaelnissenbaum** [comment]
I think you may not have understood the issue here. The problem is that the <textarea> element does not have a **visual label**. We're not discussing programmatically linked labels in this context.

**Comment by Pytal** [comment]
The idea of using
```
span
  input id="foo"
  label for="foo" id="bar"
span
textarea aria-labelledby="bar"
```
for
![image](https://github.com/nextcloud/server/assets/24800714/0c0a0508-d94a-41d2-9936-c96e596b2ce3)
was to have "Note to recipient" label both the checkbox and the textarea

Yes, this is not a separate visual label conventionally used for textarea but in this context it makes sense to use the already existing text to label the textarea as well to avoid duplication

The programmatic link then is to ensure that in this case screen readers will read out the label when focused

Does that make sense @michaelnissenbaum ?

**Comment by Pytal** [comment]
What do you think @michaelnissenbaum about the suggestion ?

**Comment by michaelnissenbaum** [comment]
I disagree with the proposed solution. The current implementation is confusing and may not be understood by some users. The <textarea> element needs a separate label.
